onnxruntime/core/platform/windows/device_discovery.cc

```diff
@@ -362,90 +362,111 @@ std::unordered_map<uint64_t, DeviceInfo> GetDeviceInfoD3D12() {
   return device_info;
 }
 
+typedef HRESULT(WINAPI* PFN_DXCoreCreateAdapterFactory)(REFIID riid, void** ppvFactory);
+
 // returns LUID to DeviceInfo
 std::unordered_map<uint64_t, DeviceInfo> GetDeviceInfoDxcore() {
   std::unordered_map<uint64_t, DeviceInfo> device_info;
 
-  // Get all GPUs and NPUs by querying WDDM/MCDM.
-  wil::com_ptr<IDXCoreAdapterFactory> adapterFactory;
-  if (FAILED(DXCoreCreateAdapterFactory(IID_PPV_ARGS(&adapterFactory)))) {
+  // Load dxcore.dll. We do this manually so there's not a hard dependency on dxcore which is newer.
```

Comment:
There is another way to solve this problem: use apiset https://learn.microsoft.com/en-us/windows/win32/apiindex/detect-api-set-availability
But I am not quite understand how it works. Because at link time the binary still need to link to something.

Reply:
As it doesn't matter if `DXCoreCreateAdapterFactory` is implemented via a stub (a stub returning nothing is the same to us in this usage as no stub being available), the value of using IsApiSetImplemented seems minimal here.